### PR TITLE
Add FXIOS-8969 [Fonts] Update: Fonts related to Home cells to use FXFontStyles

### PR DESCRIPTION
**Pull request by NeoKolian**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19806)

## :bulb: Description
fonts in following classes updated: 
JumpBackInCell,
SyncedTabCell,
RecentlySavedCell,
TopSiteItemCell.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by NeoKolian** [comment]
**before:**
Снимок экрана 2024-04-16 в 19 06 18

**Comment by NeoKolian** [comment]
**after:**
Снимок экрана 2024-04-16 в 19 06 49

File: firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift [hunk 1]
```diff
@@ -50,14 +49,12 @@ class SyncedTabCell: UICollectionViewCell, ReusableCell {
...
-        label.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .headline,
-                                                            size: UX.cardTitleFontSize)
+        label.font = FXFontStyles.Regular.headline.scaledFont()
         label.accessibilityIdentifier = AccessibilityIdentifiers.FirefoxHomepage.SyncedTab.cardTitle
     }
 
     private let syncedTabsButton: UIButton = .build { button in
-        button.titleLabel?.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .subheadline,
-                                                                         size: UX.deviceSourceFontSize)
+        button.titleLabel?.font = FXFontStyles.Regular.subheadline.scaledFont().withSize(UX.deviceSourceFontSize)
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift:57]: thanks for the changes @NeoKolian! We can remove `.withSize` and follow the existing fonts in our design system even if the font is slightly different from existing. We can have our designer confirm whether we want to follow the `subheadline` style type or use `caption1` (size 12). 

Could you remove the `withSize` and provide updated screenshots for @cwzilla to review? Thank you!

Comment (NeoKolian) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift:57]: Hello @cyndichin ! Sure, you can check new screenshots below. By the way, shall I remove `.withSize` in `RecentlySavedCell.swift` for the `itemTitle` line 29?

Comment (NeoKolian) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift:57]: ** withSize**
Снимок экрана 2024-04-17 в 16 58 24
**just scaledFont()**
Снимок экрана 2024-04-17 в 17 00 42

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: @NeoKolian Yes, let's please remove all cases `.withSize` and can have @cwzilla the screenshots once updated thanks! Also, just realize there was another PR that addressed the `JumpBackInCell` so may need to fix conflicts for that, apologies for that!

**Review by cyndichin — COMMENTED** [review]
thanks @NeoKolian for the updates! Code looks good from my side aside from a minor cleanup! Will wait for @cwzilla to confirm the font updates on the latest screenshot.

File: firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift [hunk 2]
```diff
@@ -34,9 +34,7 @@ class SyncedTabCell: UICollectionViewCell, ReusableCell {
         static let syncedDeviceImageSize = CGSize(width: 24, height: 24)
         static let tabStackTopAnchorConstant: CGFloat = 72
         static let tabStackTopAnchorCompactPhoneConstant: CGFloat = 24
-        static let cardTitleFontSize: CGFloat = 17
         static let itemTitleFontSize: CGFloat = 15
```
[the remark this answers is not in the document]
Comment (NeoKolian) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: oops sorry, done!

File: firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift (hunk 1 above)
Comment (cwzilla) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: These look good, thanks! Curious, what does scaledFont() mean?

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: @cwzilla Those are some technical details on the code side. `.withSize` was the previous implementation, but we should go with the `scaledFont`. You can see some fonts differ now with the adjustments (i.e. `Recently Saved` cards title are large). Please use the last screenshot to validate. Thanks!

Comment (cwzilla) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: @cyndichin @NeoKolian gotcha, can we adjust the recently saved cards title? they should use caption1. thanks!

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: @NeoKolian wanted to follow up on this change, let me know if you need anything from my side!

Comment (NeoKolian) [reply firefox-ios/Client/Frontend/Home/JumpBackIn/Cell/SyncedTabCell.swift]: Hello @cwzilla sorry for long response, here is the `caption1`!

Снимок экрана 2024-05-03 в 18 06 29

**Review by cyndichin — APPROVED** [review]
code looks good from my side! will wait for @cwzilla to confirm the changes before merging.

**Comment by cwzilla** [comment]
Thanks for making the changes @NeoKolian! Looks great!

**Review by cwzilla — APPROVED** [review]

**Comment by cyndichin** [comment]
thank you @NeoKolian for working with us to make these changes. It really helps us standardize our fonts!

**Pull request merged**